src/main/java/net/oauth/jsontoken/JsonToken.java

```diff
@@ -249,15 +251,24 @@ public String getKeyId() {
    */
   public SignatureAlgorithm getSignatureAlgorithm() {
     if (header == null) {
-      throw new IllegalStateException("JWT has no algorithm or header");
+      throw new IllegalStateException(
+          "JWT has no algorithm or header",
+          new InvalidJsonTokenException(ErrorCode.INTERNAL_ERROR));
```

Comment:
I added anew INVALID_HEADER error code for cases like here (https://github.com/google/jsontoken/pull/41/commits/ee4adfa09ac343f8166eff895a835991faa33c59#diff-a03b414de1be4ba95efce16ae9876f51L263). 

However, for cases where the header is missing in getSignatureAlgorithm or getHeader, would using InvalidJsonTokenException still make sense? These error cases would be similar to having IllegalStateException as a cause when signing without providing a signer. The only case these null header exception would occur would be if the constructor used for parsing is used to construct a signing JsonToken or if a null header is passed in to the signing constructor. In either case, it would be an incorrect use of the library.